jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/AppengineApiConfiguration.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.apphosting.vmruntime.jetty9;
+
+import org.eclipse.jetty.util.URIUtil;
+import org.eclipse.jetty.util.resource.Resource;
+import org.eclipse.jetty.webapp.AbstractConfiguration;
+import org.eclipse.jetty.webapp.WebAppClassLoader;
+import org.eclipse.jetty.webapp.WebAppContext;
+
+import java.util.logging.Level;
+
+public class AppengineApiConfiguration extends AbstractConfiguration {
+
+  // A class to check if the GAE API is available.
+  public static final String GAE_CHECK_CLASS = "com.google.appengine.api.ThreadManager";
+
+  // Hide the all container classes from the webapplication
+  // TODO update to use '.' when supported by Jetty
+  private static final String[] SERVER_CLASSES = {
+    "com.", 
+    "javax.", 
+    "org.", 
+    "mozilla."
+  };
+
+  private static final String[] SHARED_CLASSES = {
+    // Expose the GAE API classes
+    "com.google.appengine.api.LifecycleManager",
+    "com.google.apphosting.api.ApiProxy",
+    "com.google.apphosting.api.ApiStats",
+    "com.google.apphosting.api.CloudTrace",
+    "com.google.apphosting.api.CloudTraceContext",
+    "com.google.apphosting.api.DeadlineExceededException",
+    "com.google.apphosting.runtime.SessionData",
+    "com.google.apphosting.runtime.UncatchableError",
+
+    // Expose the standard APIs that are provided by the container
+    "javax.servlet.",
```

Comment:
are there any javax.* packages that are not part of the JDK (such as javax.servlet) that
we bundle with jetty and don't want to expose? Oh, one is probably javax.cache.

So, I wonder if it would not be simpler just to mention the specific javax packages in the SERVER_CLASSES and then I assume you don't need to specify them individually in SHARED_CLASSES?